metaverse_modules/nameplate/model.js

```diff
@@ -0,0 +1,64 @@
+import {Box3} from 'three';
+import {Geometry} from 'three/examples/jsm/deprecated/Geometry';
+
+const addToMaterials = (materials, newMaterial) => {
+  for (let i = 0; i < materials.length; i++) {
+    const mat = materials[i];
+    if (mat.name === newMaterial.name) {
+      return [materials, i];
+    }
+  }
+  materials.push(newMaterial);
+  return [materials, materials.length - 1];
+};
+
+export const getModelGeoMat = model => {
+  const newGeometry = new Geometry();
```

Comment:
The main point of instancing is to use a single draw call. But here we are using multiple materials anyway.

Can it be a single material model and we can get rid of this loader hack?

Reply:
Merging materials is only possible in very specific situations. The materials basically need to be identical, except for the 'color'.
So the correct single material can't be got when merging the materials, I think.

Reply:
@ultrabot2022 if the model has more than one material we should remove that. Basically, we should assume that everyone has the same nameplate with a single mesh and single material